go.mod

```diff
@@ -18,20 +18,21 @@ require (
 	github.com/stretchr/testify v1.8.0
 	golang.org/x/perf v0.0.0-20220722155240-3d85ee92886d
 	gopkg.in/yaml.v3 v3.0.1
-	k8s.io/api v0.24.4
-	k8s.io/apimachinery v0.24.4
+	k8s.io/api v0.25.2
+	k8s.io/apimachinery v0.25.2
 	k8s.io/autoscaler/vertical-pod-autoscaler v0.11.0
-	k8s.io/client-go v0.24.4
+	k8s.io/client-go v0.25.2
+	k8s.io/component-base v0.25.2
 	k8s.io/klog/v2 v2.70.1
 	k8s.io/sample-controller v0.24.4
```

Comment:
could you also bump that one to v0.25.2, just to make sure that all the k8s deps are in line

Reply:
actually, I forgot that this one was in flight: https://github.com/kubernetes/kube-state-metrics/pull/1819, if you rebase on master you should be able to remove the dependency bump from this PR alltogether.

Reply:
Done.